commands/project/dev/unifiedFlow.ts

```diff
@@ -86,47 +88,74 @@ export async function unifiedProjectDevFlow(
     targetTestingAccountId = profileConfig.accountId;
+  } else if (args.providedAccountId) {
+    // By pass the prompt if the user explicitly provides an --account flag.
+    targetTestingAccountId = targetProjectAccountId;
   } else {
-    const devAccountPromptResponse =
-      await selectDeveloperTestTargetAccountPrompt(accounts!, accountConfig);
-
-    targetTestingAccountId = devAccountPromptResponse.targetAccountId;
-
-    if (!!devAccountPromptResponse.notInConfigAccount) {
-      // When the developer test account isn't configured in the CLI config yet
-      // Walk the user through adding the account's PAK to the config
-      await useExistingDevTestAccount(
-        env,
-        devAccountPromptResponse.notInConfigAccount
-      );
-    } else if (devAccountPromptResponse.createNestedAccount) {
-      // Create a new developer test account and automatically add it to the CLI config
-      targetTestingAccountId = await createDeveloperTestAccountForLocalDev(
-        targetProjectAccountId,
-        accountConfig,
-        env
-      );
+    logger.log('');
+    uiLine();
+    logger.log(
+      i18n(`commands.project.subcommands.dev.logs.accountTypeInformation`)
+    );
+    logger.log('');
+    logger.log(
+      i18n(`commands.project.subcommands.dev.logs.learnMoreMessage`, {
+        learnMoreLink: uiLink(
+          i18n(`commands.project.subcommands.dev.logs.learnMoreLink`),
+          'https://developers.hubspot.com/docs/getting-started/account-types'
+        ),
+      })
+    );
+    uiLine();
+    logger.log('');
+
+    const accountType = await selectAccountTypePrompt(accountConfig);
+
+    if (accountType === HUBSPOT_ACCOUNT_TYPES.DEVELOPER_TEST) {
+      const devAccountPromptResponse =
+        await selectDeveloperTestTargetAccountPrompt(accounts!, accountConfig);
+
+      targetTestingAccountId = devAccountPromptResponse.targetAccountId;
+
+      if (!!devAccountPromptResponse.notInConfigAccount) {
+        // When the developer test account isn't configured in the CLI config yet
+        // Walk the user through adding the account's PAK to the config
+        await useExistingDevTestAccount(
+          env,
+          devAccountPromptResponse.notInConfigAccount
+        );
+      } else if (devAccountPromptResponse.createNestedAccount) {
+        // Create a new developer test account and automatically add it to the CLI config
+        targetTestingAccountId = await createDeveloperTestAccountForLocalDev(
+          targetProjectAccountId,
+          accountConfig,
+          env
+        );
+      }
+    } else if (accountType === HUBSPOT_ACCOUNT_TYPES.DEVELOPMENT_SANDBOX) {
+      const sandboxAccountPromptResponse =
+        await selectSandboxTargetAccountPrompt(accounts!, accountConfig);
+
+      targetTestingAccountId = sandboxAccountPromptResponse.targetAccountId;
+    } else {
+      targetTestingAccountId = targetProjectAccountId;
```

Comment:
Non-blocking nit:  This section might benefit from a refactor to split this up into smaller pieces.

Reply:
I may hold off and let Camden break this up however he wants to when he gets back. He was another WIP pr that touches this code too.